lib/checks/forms/submit-button.js

```diff
@@ -0,0 +1,24 @@
+let submitButtons = axe.utils.querySelectorAll(
+  virtualNode,
+  'input[type="submit"]:not([disabled]), input[type="image"]:not([disabled]), button:not([type="reset"],[type="button"],[disabled])'
+)
+
+if (submitButtons.length) {
+  return true;
+}
+
+let submitAllButtons = axe.utils.querySelectorAll(
+  virtualNode,
+  'input[type="submit"], input[type="image"], button:not([type="reset"],[type="button"])'
+)
+
+let inputNodes = axe.utils.querySelectorAll(
+  virtualNode,
+  'input'
+)
+
+if (inputNodes.length === 1 && !submitAllButtons.length) {
```

Comment:
I have a number or problems with this implementation:

1. This rule should consider the type of the input element. type=checkbox can't be submitted this way for example.
2. This rule should consider other form fields. If this form has select or textarea in it, this case shouldn't pass.
3. This rule should consider forms without form fields
4. This rule consider ARIA form fields in addition to native ones